src/components/application_manager/test/commands/mobile/add_sub_menu_request_test.cc

```diff
@@ -195,18 +235,43 @@ TEST_F(AddSubMenuRequestTest, Run_NotValidSubMenuName_UNSUCCESS) {
 }
 
 TEST_F(AddSubMenuRequestTest, Run_SUCCESS) {
-  MessageSharedPtr command_msg(CreateMessage(smart_objects::SmartType_Map));
-  (*command_msg)[am::strings::msg_params][am::strings::menu_id] = kMenuId;
-  (*command_msg)[am::strings::params][am::strings::connection_key] =
+  (*command_msg_)[am::strings::msg_params][am::strings::menu_id] = kMenuId;
+  (*command_msg_)[am::strings::params][am::strings::connection_key] =
+      kConnectionKey;
+  (*command_msg_)[am::strings::msg_params][am::strings::menu_name] =
+      "valid_sub_menu_name";
+  (*command_msg_)[am::strings::msg_params][am::strings::position] =
+      "test_position";
+
+  AddSubMenuPtr command(CreateCommand<AddSubMenuRequest>(command_msg_));
+
+  EXPECT_CALL(mock_app_manager_, application(kConnectionKey))
+      .WillOnce(Return(app));
+  EXPECT_CALL(*app, FindSubMenu(kMenuId))
+      .WillOnce(Return(static_cast<SmartObject*>(NULL)));
+
+  MessageSharedPtr result_msg(CatchHMICommandResult(CallRun(*command)));
+  const hmi_apis::FunctionID::eType kReceivedResult =
+      static_cast<hmi_apis::FunctionID::eType>(
+          (*result_msg)[am::strings::params][am::strings::function_id].asInt());
+  EXPECT_EQ(hmi_apis::FunctionID::UI_AddSubMenu, kReceivedResult);
+}
+
+TEST_F(AddSubMenuRequestTest, Run_MenuIconCorrectName_SUCCESS) {
+  (*command_msg_)[am::strings::msg_params][am::strings::menu_id] = kMenuId;
```

Comment:
@VVeremjova I see same line of code about filling message in all test cases may be move this code to separate method.